native/native.tests/tests/org/jetbrains/kotlin/konan/test/blackbox/ProgramNameTest.kt

```diff
@@ -0,0 +1,66 @@
+package org.jetbrains.kotlin.konan.test.blackbox
+
+import org.jetbrains.kotlin.konan.test.blackbox.support.*
+import org.jetbrains.kotlin.konan.test.blackbox.support.compilation.TestCompilationResult.Companion.assertSuccess
+import org.jetbrains.kotlin.konan.test.blackbox.support.util.ClangDistribution
+import org.jetbrains.kotlin.konan.test.blackbox.support.util.compileWithClang
+import org.jetbrains.kotlin.native.executors.runProcess
+import org.junit.jupiter.api.Tag
+import org.junit.jupiter.api.Test
+import java.io.File
+import kotlin.test.assertEquals
+import kotlin.time.Duration.Companion.seconds
+
+@Tag("program-name")
+class ProgramNameTest : AbstractNativeSimpleTest() {
+
+    @Test
+    fun programNameTest() {
+        // 1. Compile main.c to main.cexe
+
+        val cExecutable = buildDir.resolve("main.cexe")
+        compileWithClang(
+            clangDistribution = ClangDistribution.Llvm,
+            sourceFiles = listOf(sourceDir.resolve("main.c")),
+            outputFile = cExecutable,
+            additionalClangFlags = listOf("-Wall", "-Werror"),
+        ).assertSuccess()
+
+        // 2. Compile kotlinPrintEntryPoint.kt to kotlinPrintEntryPoint.kexe
+
+        val kotlinCompilation = compileToExecutableInOneStage(
+            generateTestCaseWithSingleFile(
+                sourceFile = sourceDir.resolve("kotlinPrintEntryPoint.kt"),
+                testKind = TestKind.STANDALONE_NO_TR,
+                extras = TestCase.NoTestRunnerExtras("main")
+            )
+        ).assertSuccess()
+
+        // 3. run main.cexe (with different parameters) to call kotlin executable
+
+        fun validate(expected: String, vararg args: String) {
+            val binaryName = kotlinCompilation.resultingArtifact.executableFile.path
+            val result = runProcess(cExecutable.absolutePath, binaryName, *args) {
+                timeout = 60.seconds
+            }
+            assertEquals("calling exec...\n$expected", result.stdout)
```

Comment:
I suspect this code should be fixed to handle `\r\n` on Windows.

Reply:
Good catch, thank you again! The assertion is now platform agnostic due to additional sanitization.